detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/ConfigPropertySpec.kt

```diff
@@ -145,6 +146,56 @@ class ConfigPropertySpec {
                     }
                 }
             }
+
+            @Nested
+            inner class `ExplainedValues property` {
+                private val defaultValue = explainedValues("aValue" to "aReason")
+
+                @Nested
+                inner class `value defined as list` {
+                    private val subject = object : TestConfigAware("present" to listOf("a", "b", "c")) {
+                        val present: ExplainedValues by config(defaultValue)
+                        val notPresent: ExplainedValues by config(defaultValue)
+                    }
+
+                    @Test
+                    fun `uses the value provided in config if present`() {
+                        assertThat(subject.present)
+                            .extracting(ExplainedValue::value, ExplainedValue::reason)
+                            .containsExactly(tuple("a", null), tuple("b", null), tuple("c", null))
+                    }
+
+                    @Test
+                    fun `uses the default value if not present`() {
+                        assertThat(subject.notPresent).isEqualTo(defaultValue)
+                    }
+                }
+
+                @Nested
+                inner class `value defined as list of maps` {
+                    private val subject = object : TestConfigAware(
+                        "present" to listOf(
+                            mapOf("value" to "a", "reason" to "reasonA"),
+                            mapOf("value" to "b", "reason" to null)
```

Comment:
Should we handle something like this too?
```suggestion
                            mapOf("value" to "b"), // missing reason
```
I think that usually we will find this case more often than having a `null` reason.

Reply:
Good call. I added this case